PukingPredator/Assets/Scripts/Player.cs

```diff
@@ -32,34 +28,32 @@ private bool canJump
         get => isGrounded && !isJumpOnCooldown;
     }
 
+    private float pukeDistance = 2f;
+
     private void Start()
     {
         gameInput.onEatAction += GameInput_OnEat;
         gameInput.onPukeAction += GameInput_OnPuke;
         gameInput.onResetLevelAction += ResetLevel;
-        //rigidBody = transform.Find("Capsule").GetComponent<Rigidbody>();
         rigidBody = GetComponent<Rigidbody>();
     }
 
     private void GameInput_OnPuke(object sender, System.EventArgs e)
     {
-        Item itemToPlace = inventory.RemoveItem();
-        if (itemToPlace != null)
-        {
-            Debug.Log(lastDir);
-            itemToPlace.MoveItem(transform.position + lastDir);
-            itemToPlace.PlaceItem();
-        }
+        Item itemToPlace = inventory.PopItem();
+
+        if (itemToPlace is null) { return; }
+
+        var pukeDir = transform.forward;
+        itemToPlace.PlaceAt(transform.position + pukeDir*pukeDistance);
     }
 
     private void GameInput_OnEat(object sender, System.EventArgs e)
     {
-        if (inventory.isFull())
-        {
-            return;
-        }
+        if (inventory.isFull) { return; }
+
         // Define the ray, starting from the player's position, shooting forward
-        Ray ray = new Ray(transform.position, lastDir);
+        Ray ray = new Ray(transform.position, transform.forward);
```

Comment:
good optimization just double test if eating still works, like eating while moving and eating while stationary

Reply:
tested that and it seemed to work, only case to worry about down the road is if the player rotates around other axis than just Z, but in that case we will just need to project the forward vector back to the XY plane and then renormalize it. this version works a bit better than the previous one though because the last one would make you eat stuff in the direction of your input, not where the character is facing, which was visually unintuitive.